### PR TITLE
[prometheus-pushgateway] Ensure PDB labels are same as the matchLabels of the owner app

### LINKED ISSUE #2665: [prometheus-pushgateway] PDB selector labels do not uniquely single out pods of the helm release
### Describe the bug a clear and concise description of what the bug is.

If you have multiple releases of prometheus-pushgateway in a single namespace, the PDB labels cannot differentiate between the pods of the different releases

### What's your helm version?

version.BuildInfo{Version:"v3.7.1", GitCommit:"1d11fcb5d3f3bf00dbe6fe31b8412839a96b3dc4", GitTreeState:"clean", GoVersion:"go1.16.9"}

### What's your kubectl version?

Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.12", GitCommit:"c6939792865ef0f70f92006081690d77411c8ed5", GitTreeState:"clean", BuildDate:"2022-09-21T12:20:29Z", GoVersion:"go1.17.13", Compiler:"gc", Platform:"darwin/arm64"}

### Which chart?

prometheus-pushgateway

### What's the chart version?

1.20.1

### What happened?

_No response_

### What you expected to happen?

_No response_

### How to reproduce it?

_No response_

### Enter the changed values of values.yaml?

```yaml
podDisruptionBudget:
  maxUnavailable: 1
```

### Enter the command that you execute and failing/misfunctioning.

helm template prometheus-pushgateway-1 prometheus-community/prometheus-pushgateway --namespace kube-prometheus-stack --set podDisruptionBudget.maxUnavailable=1 -s templates/pdb.yaml
helm template prometheus-pushgateway-2 prometheus-community/prometheus-pushgateway --namespace kube-prometheus-stack --set podDisruptionBudget.maxUnavailable=1 -s templates/pdb.yaml

The value of spec.selector.matchLabels should be different in the 2 cases

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by mikhailadvani**
#### What this PR does / why we need it

To support multiple releases of pushgateway in a single namespace.

#### Which issue this PR fixes

- fixes #2665 

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by monotek — APPROVED** [review]

File: charts/prometheus-pushgateway/templates/deployment.yaml [hunk 1]
```diff
@@ -13,8 +13,7 @@ spec:
   {{- end }}
   selector:
     matchLabels:
-      app: {{ template "prometheus-pushgateway.name" . }}
-      release: {{ .Release.Name }}
+{{- include "prometheus-pushgateway.selectorLabels" . | indent 6 }}
```
Comment (zanhsieh) [inline charts/prometheus-pushgateway/templates/deployment.yaml:16]:
```suggestion
      {{- include "prometheus-pushgateway.selectorLabels" . | nindent 6 }}
```

File: charts/prometheus-pushgateway/templates/pdb.yaml [hunk 2]
```diff
@@ -12,6 +12,6 @@ metadata:
 spec:
   selector:
     matchLabels:
-      app: {{ template "prometheus-pushgateway.name" . }}
+{{- include "prometheus-pushgateway.selectorLabels" . | indent 6 }}
 {{ toYaml .Values.podDisruptionBudget | indent 2 }}
-{{- end -}}
\ No newline at end of file
+{{- end -}}
```
Comment (zanhsieh) [inline charts/prometheus-pushgateway/templates/pdb.yaml]:
```suggestion
      {{- include "prometheus-pushgateway.selectorLabels" . | nindent 6 }}
  {{- toYaml .Values.podDisruptionBudget | nindent 2 }}
{{- end -}}
```

File: charts/prometheus-pushgateway/templates/_helpers.tpl [hunk 3]
```diff
@@ -65,6 +65,14 @@ Return the appropriate apiVersion for networkpolicy.
 {{- end -}}
 {{- end -}}
 
+{{/*
+Selector labels
+*/}}
+{{- define "prometheus-pushgateway.selectorLabels" }}
+app: {{ template "prometheus-pushgateway.name" . }}
+release: {{ .Release.Name }}
+{{- end }}
```
Comment (zanhsieh) [inline charts/prometheus-pushgateway/templates/_helpers.tpl:74]: Align with previous _helpers.tpl
```suggestion
{{- define "prometheus-pushgateway.selectorLabels" -}}
app: {{ template "prometheus-pushgateway.name" . }}
release: {{ .Release.Name }}
{{- end -}}
```

File: charts/prometheus-pushgateway/templates/statefulset.yaml [hunk 4]
```diff
@@ -10,8 +10,7 @@ spec:
   serviceName: {{ template "prometheus-pushgateway.fullname" . }}
   selector:
     matchLabels:
-      app: {{ template "prometheus-pushgateway.name" . }}
-      release: {{ .Release.Name }}
+{{- include "prometheus-pushgateway.selectorLabels" . | indent 6 }}
```
Comment (zanhsieh) [inline charts/prometheus-pushgateway/templates/statefulset.yaml:13]:
```suggestion
      {{- include "prometheus-pushgateway.selectorLabels" . | nindent 6 }}
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

File: charts/prometheus-pushgateway/templates/statefulset.yaml [hunk 5]
```diff
@@ -25,7 +24,7 @@ spec:
     - metadata:
         {{- if .Values.persistentVolume.annotations }}
         annotations:
-      {{ toYaml .Values.persistentVolume.annotations | indent 10 }}
+          {{ toYaml .Values.persistentVolume.annotations | nindent 10 }}
         {{- end }}
```
Comment (zanhsieh) [inline charts/prometheus-pushgateway/templates/statefulset.yaml:28]:
```suggestion
        {{- with .Values.persistentVolume.annotations }}
        annotations:
          {{- toYaml . | nindent 10 }}
        {{- end }}
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

**Review by zanhsieh — APPROVED** [review]

**Review by zanhsieh — APPROVED** [review]

**Review by monotek — APPROVED** [review]

**Pull request merged**